File: bindings/jvm/sone-core/src/main/java/dev/sone/Font.java
package dev.sone;

import java.util.List;

/**
 * Font registration on the process-wide engine, for programs that do not want
 * to own one.
 *
 * <p>Skia carries no system fonts, so at least one family must be registered
 * before any text renders.
 */
public final class Font {

    private Font() {
    }

    public static void load(String name, String path) {
        Engine.getDefault().registerFontFile(name, path);
    }

    public static void load(String name, byte[] data) {
        Engine.getDefault().registerFont(name, data);
    }

    public static boolean has(String name) {
        return Engine.getDefault().hasFont(name);
    }

    public static List<String> families() {
        return Engine.getDefault().fontFamilies();
    }

    public static void reset() {
        Engine.getDefault().resetFonts();
    }
}

File: bindings/csharp/src/Sone/Font.cs
namespace Sone;

/// <summary>
/// Font registration on the process-wide engine, for scripts that do not want
/// to create one. Everything here is <see cref="Engine.Default"/>.
/// </summary>
/// <remarks>
/// Skia carries no system fonts, so at least one family must be registered
/// before rendering any text.
/// </remarks>
public static class Font
{
    /// <summary>Register a family from a TTF/OTF file.</summary>
    public static void Load(string name, string path) =>
        Engine.Default.RegisterFontFile(name, path);

    /// <summary>Register a family from raw TTF/OTF bytes.</summary>
    public static void Load(string name, ReadOnlySpan<byte> data) =>
        Engine.Default.RegisterFont(name, data);

    /// <summary>Whether a family has been registered.</summary>
    public static bool Has(string name) => Engine.Default.HasFont(name);

    /// <summary>Every registered family name.</summary>
    public static IReadOnlyList<string> Families() => Engine.Default.FontFamilies();

    /// <summary>Drop every registered font.</summary>
    public static void Reset() => Engine.Default.ResetFonts();
}
